### DISCUSSION TITLE
Vue `arrow` missing on MiddlewareData?

### DISCUSSION THREAD

**Discussion by 9mm** [floating-ui/floating-ui#2103] [Q&A]
I'm following the vue docs which are a bit sparse, it also was quite confusing because I needed to add `.value` to x, y, etc (wasn't in the docs, took me quite awhile to find that).

I notice the `arrow` docs for vue make it seem like it just works. It doesnt seem to update so I looked at the normal docs and it seems I need to manually set the style with `middlewareData.arrow`

When I look at the value of `middlewareData` it is just `{}`:

Does anyone have any ideas how to get the arrow working?

```javascript
  setup() {
    const anchor = ref(null);
    const floating = ref(null);
    const point = ref(null);

    const middleware = ref([
      offset(10),
      autoPlacement(),
      // shift(),
      arrow({ element: point }),
    ]);

    const {
      x, y, strategy,
    } = useFloating(anchor, floating, {
      whileElementsMounted: autoUpdate,
      placement: 'top',
      middleware,
    });

    const style = computed(() => ({
      position: strategy.value,
      top: `${y.value ?? 0}px`,
      left: `${x.value ?? 0}px`,
      width: 'max-content',
    }));

    // const { x: xA, y: yA } = middlewareData.arrow;

    // const arrowStyle = computed(() => ({
    //   left: `${xA ?? 0}px`,
    //   top: `${yA ?? 0}px`,
    // }));

    return {
      anchor,
      floating,
      point,
      style,
      //arrowStyle,
    };
  },

```

**Comment by atomiks** [comment]
Should it be `middlewareData.value.arrow`? @lozinsky

**Reply by lozinsky** [reply]
Yes, that's right.

```js
const { middlewareData } = useFloating(
  reference,
  floating,
  {
    middleware: [arrow({ element: floatingArrow })],
  }
);
const floatingArrowX = computed(() => middlewareData.value.arrow?.x ?? null);
const floatingArrowY = computed(() => middlewareData.value.arrow?.y ?? null);
```

**Comment by 9mm** [comment]
I tried that and middlewareData.value just returns an empty object {}

**Reply by lozinsky** [reply]
It returns an empty object initially or when no `reference`, `floating` or `arrow` elements are rendered. I made a basic example of how to use `middlewareData`: https://codesandbox.io/p/sandbox/floating-ui-vue-arrow-b0m4vi?file=/src/App.vue

**Reply by 9mm** [reply]
Interesting... I guess I was extremely tired, thank you. Somehow I left the middlewareData out of the computed property 🤦🏻‍♂️

Thanks Lozinksy!

**Reply by 9mm** [reply]
Do you happen to know off the top of your head how I get the arrow to appear on the proper side when using `autoPlacement()`? I will look through the docs after I fix the rest of this, just thought you might know.

**Reply by atomiks** [reply]
`placement` returned from `useFloating()` is the final placement. Did you try looking in the arrow docs for it? It seems many get confused by this part.

**Reply by 9mm** [reply]
Ok, I officially got it working I think. That more grueling than expected, but I understand why it's like this. at least it's done now. The full control is nice for bigger projects but for smaller ones 😵‍💫

```javascript
  setup() {
    const anchor = ref(null);
    const floating = ref(null);
    const floatingArrow = ref(null);
    const offsetAmount = 10;

    const middleware = ref([
      autoPlacement(),
      offset(offsetAmount),
      arrow({ element: floatingArrow }),
    ]);

    const {
      x, y, strategy, placement, middlewareData,
    } = useFloating(anchor, floating, {
      whileElementsMounted: autoUpdate,
      middleware,
    });

    const style = computed(() => ({
      position: strategy.value,
      top: `${y.value ?? 0}px`,
      left: `${x.value ?? 0}px`,
      width: 'max-content',
    }));

    /* arrow */

    const arrowX = computed(() => middlewareData.value.arrow?.x ?? null);
    const arrowY = computed(() => middlewareData.value.arrow?.y ?? null);

    const arrowStyle = computed(() => {
      const side = placement.value.split('-')[0];
      const staticSide = {
        top: 'bottom', right: 'left', bottom: 'top', left: 'right',
      }[side];

      return {
        left: arrowX.value != null ? `${arrowX.value}px` : '',
        top: arrowY.value != null ? `${arrowY.value}px` : '',
        [staticSide]: `${-offsetAmount}px`,
      };
    });

    return {
      anchor,
      floating,
      floatingArrow,
      style,
      arrowStyle,
    };
```

**Reply by atomiks** [reply]
Yeah the positioning part of the library is more like CSS, you set everything up without any helpers to do it. 

There's an open issue about making arrows easier to create (which is by far the most painful part). Styling the floating element is straightforward but the arrow is annoying for sure.

**Reply by 9mm** [reply]
Thanks @atomiks keep up the awesome work!
